### DISCUSSION TITLE
Module Settings Secure?

### DISCUSSION THREAD

**Discussion by vnomus** [oqtane/oqtane.framework#3754] [Q&A]
Are module settings secure? How can we create/store/manage module level settings (e.g. module-level connection strings) in a secure fashion? Are there any examples you could point us to? In this case we' need the setting value available in the Client code.

**Comment by leigh-pointer** [comment]
@vnomus if you look at the site setting you will see settings can be saved with a flag.

**Comment by sbwalker** [comment]
Settings have an IsPrivate flag which indicates whether or not they can be shared with a client application via the Settings API or if they are sensitive and must remain on the server only. Note that the API also contains logic so that if a user is an Administrator they can access the Settings on the client or server (for administrative purposes). Items such as connection strings should be set as IsPrivate as you would never want to disclose this information publicly.
